packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -231,6 +231,60 @@ class _ProductListPageState extends State<ProductListPage>
         actionModeTitle: Text(appLocalizations.compare_products_appbar_title),
         actionModeSubTitle:
             Text(appLocalizations.compare_products_appbar_subtitle),
+        actionModeActions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.delete),
+            onPressed: () async {
+              if (_selectedBarcodes.isNotEmpty) {
+                await showDialog<bool>(
+                  context: context,
+                  builder: (BuildContext context) {
+                    return SmoothAlertDialog(
+                      body: Text(
+                        appLocalizations.confirm_clear_selected_user_list,
+                      ),
+                      positiveAction: SmoothActionButton(
+                        onPressed: () async {
+                          await daoProductList.bulkSet(
+                            productList,
+                            _selectedBarcodes.toList(growable: false),
+                            include: false,
+                          );
+                          await daoProductList.get(productList);
+                          setState(() {});
```

Comment:
A `setState` before your check to `mounted` is meaningless, because it could fail.
You should flip the two calls